src/System.Private.CoreLib/src/System/Reflection/Emit/DynamicILGenerator.cs

```diff
@@ -717,14 +717,21 @@ private static int CalculateNumberOfExceptions(__ExceptionInfo[] excp)
             stackSize = m_stackSize;
             if (m_exceptionHeader != null && m_exceptionHeader.Length != 0)
             {
-                if (m_exceptionHeader.Length < 4)
+                // If length < 4, fail. Below pattern elides future bounds checks in method.
+                if (3 >= (uint)m_exceptionHeader.Length)
                     throw new FormatException();
 
                 byte header = m_exceptionHeader[0];
 
                 if ((header & 0x40) != 0) // Fat
                 {
-                    EHCount = (BinaryPrimitives.ReadInt32LittleEndian(m_exceptionHeader.AsSpan(1)) - 4) / 24;
+                    Span<byte> size = stackalloc byte[4];
+                    size[0] = m_exceptionHeader[1]; // n.b. this is a 24-bit integer, not a 32-bit integer
+                    size[1] = m_exceptionHeader[2];
+                    size[2] = m_exceptionHeader[3];
+                    size[3] = 0;
+
+                    EHCount = (BitConverter.ToInt32(size) - 4) / 24;
```

Comment:
This does not look right wrt. endianess.

Reply:
Here's the original code before yesterday's change:
https://github.com/dotnet/coreclr/blob/8913aaa1388e263d537b19e78caaed6b124fc0ea/src/System.Private.CoreLib/src/System/Reflection/Emit/DynamicILGenerator.cs#L726-L729

The original code assumed machine-endianness (as does this PR), and the modified code introduced recently assumes little-endian. I can update this to assume little-endian if desired.

Reply:
Yes, this should be little endian.

Reply:
Now that I typed that, I realize this wouldn't make sense in big-endian. Since the last byte would always be 0, it means the _EHCount_ value would always be too large. Surely this code assumes little-endian.